### PR TITLE
Quality Declarations for rcl_action, rcl_lifecycle, yaml_parser

### PR DESCRIPTION
**Pull request by brawner**
As part of the effort to push to Quality Level 1, this adds quality declarations for rcl_action, rcl_lifecycle, and rcl_yaml_param_parser.

Signed-off-by: Stephen Brawner <brawner@gmail.com>

### CODE REVIEW

File: rcl_lifecycle/QUALITY_DECLARATION.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+This package uses DCO as its confirmation of contributor origin policy. More information can be found in [CONTRIBUTING](../CONTRIBUTING.md).
+
+### Peer Review Policy [2.iii]
+
+Following the recommended guidelines for ROS Core packages, all pull requests must have at least 1 peer review.
+
+### Continuous Integration [2.iv]
+
+All pull requests must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers)
```
Comment (ahcorde) [inline rcl_lifecycle/QUALITY_DECLARATION.md:51]:
```suggestion
All pull requests must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers).
```

File: rcl_lifecycle/QUALITY_DECLARATION.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+Most of `rcl_lifecycle` has embedded API documentation. It is not yet hosted publicly.
+
+### License [3.iii]
+
+The license for `rcl_lifecycle` is Apache 2.0, and a summary is in each source file, the type is declared in the [package.xml](package.xml) manifest file, and a full copy of the license is in the [LICENSE](../LICENSE) file.
+
+There is an automated test which runs a linter that ensures each file has a license statement.
+
+Most recent test results can be found [here](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_lifecycle/copyright/)
```
Comment (ahcorde) [inline rcl_lifecycle/QUALITY_DECLARATION.md:74]:
```suggestion
Most recent test results can be found [here](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_lifecycle/copyright/).
```

File: rcl_lifecycle/QUALITY_DECLARATION.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/rcl_lifecycle/)
+* [linux_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_lifecycle/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rcl_lifecycle/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rcl_lifecycle/)
+
+# Security [7]
+
+## Vulnerability Disclosure Policy [7.i]
+
+This package does not yet have a Vulnerability Disclosure Policy
```
Comment (ahcorde) [inline rcl_lifecycle/QUALITY_DECLARATION.md:141]:
```suggestion
This package does not yet have a Vulnerability Disclosure Policy.
```

File: rcl_action/QUALITY_DECLARATION.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/rcl_action/)
+* [linux_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_action/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rcl_action/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rcl_action/)
+
+# Security [7]
+
+## Vulnerability Disclosure Policy [7.i]
+
+This package does not yet have a Vulnerability Disclosure Policy
```
Comment (ahcorde) [inline rcl_action/QUALITY_DECLARATION.md:142]:
```suggestion
This package does not yet have a Vulnerability Disclosure Policy.
```

File: rcl_yaml_param_parser/QUALITY_DECLARATION.md [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+All of `rcl_yaml_param_parser` has embedded API documentation. It is not yet hosted publicly.
+
+### License [3.iii]
+
+The license for `rcl_yaml_param_parser` is Apache 2.0, and a summary is in each source file, the type is declared in the [package.xml](package.xml) manifest file, and a full copy of the license is in the [LICENSE](../LICENSE) file.
+
+There is an automated test which runs a linter that ensures each file has a license statement.
+
+Most recent test results can be found [here](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_yaml_param_parser/copyright/)
```
Comment (ahcorde) [inline rcl_yaml_param_parser/QUALITY_DECLARATION.md:73]:
```suggestion
Most recent test results can be found [here](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_yaml_param_parser/copyright/).
```

File: rcl_yaml_param_parser/QUALITY_DECLARATION.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+This package uses DCO as its confirmation of contributor origin policy. More information can be found in [CONTRIBUTING](../CONTRIBUTING.md).
+
+### Peer Review Policy [2.iii]
+
+Following the recommended guidelines for ROS Core packages, all pull requests must have at least 1 peer review.
+
+### Continuous Integration [2.iv]
+
+All pull requests must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers)
```
Comment (ahcorde) [inline rcl_yaml_param_parser/QUALITY_DECLARATION.md:51]:
```suggestion
All pull requests must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers).
```

File: rcl_yaml_param_parser/QUALITY_DECLARATION.md [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/rcl_yaml_param_parser/)
+* [linux_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_yaml_param_parser/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rcl_yaml_param_parser/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rcl_yaml_param_parser/)
+
+# Security [7]
+
+## Vulnerability Disclosure Policy [7.i]
+
+This package does not yet have a Vulnerability Disclosure Policy
```
Comment (ahcorde) [inline rcl_yaml_param_parser/QUALITY_DECLARATION.md:137]:
```suggestion
This package does not yet have a Vulnerability Disclosure Policy.
```

File: rcl_action/QUALITY_DECLARATION.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+
+This package uses DCO as its confirmation of contributor origin policy. More information can be found in [CONTRIBUTING](../CONTRIBUTING.md).
+
+### Peer Review Policy [2.iii]
+
+Following the recommended guidelines for ROS Core packages, all pull requests must have at least 1 peer review.
+
+### Continuous Integration [2.iv]
+
+All pull requests must pass CI on all [tier 1 platforms](https://www.ros.org/reps/rep-2000.html#support-tiers)
```
[the remark this answers is not in the document]
Comment (brawner) [reply rcl_action/QUALITY_DECLARATION.md]: Fixed.

File: rcl_action/QUALITY_DECLARATION.md [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+
+Most of `rcl_action` has embedded API documentation and it is [hosted](http://docs.ros2.org/latest/api/rcl_action/).
+
+### License [3.iii]
+
+The license for `rcl_action` is Apache 2.0, and a summary is in each source file, the type is declared in the [package.xml](package.xml) manifest file, and a full copy of the license is in the [LICENSE](../LICENSE) file.
+
+There is an automated test which runs a linter that ensures each file has a license statement.
+
+Most recent test results can be found [here](https://ci.ros2.org/view/nightly/job/nightly_linux_release/1525/testReport/rcl_action/copyright/)
```
[the remark this answers is not in the document]
Comment (brawner) [reply rcl_action/QUALITY_DECLARATION.md]: Fixed.

File: rcl_action/QUALITY_DECLARATION.md (hunk 4 above)
Comment (brawner) [reply rcl_action/QUALITY_DECLARATION.md]: Fixed.

File: rcl_lifecycle/QUALITY_DECLARATION.md (hunk 1 above)
Comment (brawner) [reply rcl_lifecycle/QUALITY_DECLARATION.md]: Fixed.

File: rcl_lifecycle/QUALITY_DECLARATION.md (hunk 2 above)
Comment (brawner) [reply rcl_lifecycle/QUALITY_DECLARATION.md]: Fixed.

File: rcl_lifecycle/QUALITY_DECLARATION.md (hunk 3 above)
Comment (brawner) [reply rcl_lifecycle/QUALITY_DECLARATION.md]: Fixed.

File: rcl_yaml_param_parser/QUALITY_DECLARATION.md (hunk 6 above)
Comment (brawner) [reply rcl_yaml_param_parser/QUALITY_DECLARATION.md]: Fixed.

File: rcl_yaml_param_parser/QUALITY_DECLARATION.md (hunk 5 above)
Comment (brawner) [reply rcl_yaml_param_parser/QUALITY_DECLARATION.md]: Fixed.

File: rcl_yaml_param_parser/QUALITY_DECLARATION.md (hunk 7 above)
Comment (brawner) [reply rcl_yaml_param_parser/QUALITY_DECLARATION.md]: Fixed.

File: rcl_action/QUALITY_DECLARATION.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+`rcl_action` uses and passes all the standard linters and static analysis tools for a C package as described in the [ROS 2 Developer Guide](https://index.ros.org/doc/ros2/Contributing/Developer-Guide/#linters).
+
+Results of the nightly linter tests can be found [here](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rcl_action).
+
+## Dependencies [5]
+
+### Direct Runtime ROS Dependencies [5.i]/[5.ii]
+
+`rcl_action` has the following runtime ROS dependencies:
+* `action_msgs`
```
Comment (ahcorde) [inline rcl_action/QUALITY_DECLARATION.md:115]: can we describe the dependencies as it's done [here](https://github.com/ros2/rcl/pull/639/files#diff-5738374578ac1a78c061cd6f5b8dc8f4R143)?

Comment (brawner) [reply rcl_action/QUALITY_DECLARATION.md]: Done.

File: rcl_action/QUALITY_DECLARATION.md [hunk 11]
```diff
@@ -109,17 +109,44 @@ Results of the nightly linter tests can be found [here](https://ci.ros2.org/view
...
-* `rcl`
-* `rcutils`
-* `rmw`
-* `rosidl_runtime_c`
 
-It has several "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.
-It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.
+#### `action_msgs`
+
+`action_msgs` provides the message interfaces for communication.
```
Comment (ahcorde) [inline rcl_action/QUALITY_DECLARATION.md:123]:
```suggestion
`action_msgs` messages and services for ROS 2 actions
```

File: rcl_lifecycle/QUALITY_DECLARATION.md [hunk 12]
```diff
@@ -108,17 +108,44 @@ Results of the nightly linter tests can be found [here](https://ci.ros2.org/view
...
-* `rcl`
-* `rcutils`
-* `rmw`
-* `rosidl_runtime_c`
 
-It has several "buildtool" dependencies, which do not affect the resulting quality of the package, because they do not contribute to the public library API.
-It also has several test dependencies, which do not affect the resulting quality of the package, because they are only used to build and run the test code.
+#### `lifecycle_msgs`
+
+`lifecycle_msgs` provides the message interfaces for communication.
```
Comment (ahcorde) [inline rcl_lifecycle/QUALITY_DECLARATION.md:122]:
```suggestion
`lifecycle_msgs` message and service definitions for managing lifecycle nodes.
```

Comment (brawner) [reply rcl_lifecycle/QUALITY_DECLARATION.md]: Done.

File: rcl_action/QUALITY_DECLARATION.md (hunk 11 above)
Comment (brawner) [reply rcl_action/QUALITY_DECLARATION.md]: Done.

**Pull request merged**